Weekly and job consistency scripts

fejta opened the pull request:
`weekly-consistency.sh` shows the weekly consistency for PRs
`job-flakes.sh` computes the consistency for each job

/assign @spxtr @brendandburns

fejta commented:
/unassign @spxtr 
/assign @rmmh

rmmh commented on experiment/bigquery/README.md:
```diff
@@ -12,11 +12,18 @@ instructions](https://cloud.google.com/sdk/downloads#apt-get)).
     - Uses `flakes.sql` to extract and group data from BigQuery
-    - Usage `flakes.sh | tee flakes-$(date %Y-%m-%d).json`
-    - Latest results: [flakes-latest](https://github.com/kubernetes/test-infra/blob/master/experiment/bigquery/flakes-latest.json)
+    - Usage: `flakes.sh | tee flakes-$(date %Y-%m-%d).json`
```
The usage strings should be be `./flakes.sh`

fejta replied:
Meh I disagree as that can be inferred from context and assumes that they cd into this directory.

rmmh replied:
It just saves a little time for copy-pasting.

fejta replied:
Again, meh

rmmh commented on experiment/bigquery/flakes.sql:
```diff
@@ -17,73 +17,75 @@ select
+      from (
+        SELECT
+          job,
+          regexp_extract(path, r'pull/(\d+)') as num,
```
Use the `number` column.

fejta replied:
That is the build number, which I could care less about, not the PR number

rmmh commented on experiment/bigquery/flakes.sql:
```diff
@@ -17,73 +17,75 @@ select
+    select
+      job,
+      stamp,
+      num,
```
unused?

fejta replied:
Ack

rmmh commented on experiment/bigquery/flakes.sql:
```diff
@@ -17,73 +17,75 @@ select
+      stamp,
+      num,
+      commit,
+      if(passed = runs or passed = 0, 0, 1) flaked,
```
I don't understand this boolean-- set flaked to 1 if passed is equal to 0 or runs? Why would passing every run indicate a flake?

rmmh replied:
Oh, I misread the if-- it DIDN'T flake if it always passed or was permanently broken.

fejta replied:
Ack

rmmh commented on experiment/bigquery/README.md:
```diff
@@ -12,11 +12,18 @@ instructions](https://cloud.google.com/sdk/downloads#apt-get)).
-    - Usage `flakes.sh | tee flakes-$(date %Y-%m-%d).json`
-    - Latest results: [flakes-latest](https://github.com/kubernetes/test-infra/blob/master/experiment/bigquery/flakes-latest.json)
+    - Usage: `flakes.sh | tee flakes-$(date %Y-%m-%d).json`
+    - Latest results: [flakes-latest.json](https://github.com/kubernetes/test-infra/blob/master/experiment/bigquery/flakes-latest.json)
```
A [relative link](https://github.com/blog/1395-relative-links-in-markup-files) might be more appropriate here, instead of hardcoding the Github URL.

fejta replied:
Done, thanks!

rmmh commented on experiment/bigquery/weekly-consistency.sql:
```diff
@@ -0,0 +1,42 @@
+    num,
+    commit,
+    max(if(passed == runs or passed == 0, 0, 1)) flaked,
+    max(runs) builds
```
unused

fejta replied:
Ack

fejta replied:
Done

rmmh commented:
Can you add some more comments about what (broadly) each of the sub-queries is computing, and what the final result means? Documenting precisely what "consistency" means would be valuable.

fejta commented on experiment/bigquery/flakes.sql:
```diff
@@ -17,73 +17,75 @@ select
+  from (
+    select
+      job,
+      stamp,
```
Ack

fejta commented:
Documented consistency in the README. Also added some comments. I also added TODOs around the various unused fields. I'm not at all convinced these queries are at all optimized. I'd like to clean them up later if it turns out we actually want to keep any of them.

rmmh commented:
why do you prefer `/* comment */` over `# comment` ?

/lgtm

fejta commented:
Thanks! Only because  `/* comment */` is the first thing I found. Can update these later, just want to get something under source control.
